plugin/storage/es/spanstore/reader_test.go

```diff
@@ -170,63 +171,64 @@ func TestSpanReaderIndices(t *testing.T) {
 	serviceDataLayout := "2006-01-02"
 	spanDataLayoutFormat := date.UTC().Format(spanDataLayout)
 	serviceDataLayoutFormat := date.UTC().Format(serviceDataLayout)
+	tracer := jtracer.NoOp().OTEL.Tracer("")
 
 	testCases := []struct {
 		indices []string
 		params  SpanReaderParams
 	}{
 		{
 			params: SpanReaderParams{
-				Client: client, Logger: logger, MetricsFactory: metricsFactory,
+				Client: client, Logger: logger, MetricsFactory: metricsFactory, Tracer: tracer,
```

Comment:
These tests are using a poor pattern, let's fix it. When parameterizing tests, params should only contain things that are different between test cases. These three values are always the same, they can be assigned inside the loop below.